### PR TITLE
ENIFP-2076:  added range selector

### PR DESCRIPTION
**Pull request by DoChunyu**
added range selector

### CODE REVIEW

File: packages/es-components/src/components/controls/range-selector/RangeSelector.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    if (currentMinValue > maxValue) return 100;
+    return ((currentMinValue - minValue) / (maxValue - minValue)) * 100;
+  };
+  const initStateMax = () => {
+    if (currentMaxValue > maxValue) return 100;
+    if (currentMaxValue < minValue) return 0;
+    return ((currentMaxValue - minValue) / (maxValue - minValue)) * 100;
+  };
+  const [min, setMin] = useState(currentMinValue);
+  const [max, setMax] = useState(currentMaxValue);
```
Comment (stevematney) [inline packages/es-components/src/components/controls/range-selector/RangeSelector.js:78]: This kind of component is going to be so specific to its implementation, that I think it should be a [Controlled Component](https://legacy.reactjs.org/docs/forms.html#controlled-components). This would allow us to eliminate any internal state, and simply use the `onChange` event to tell our parent what the current min and max are.

An implementation of something like that might be:
```jsx
const [myRange, setMyRange] = useState([1, 10]);
<RangeSelector
  min={1}
  max={10}
  currentMin={myRange[0]}
  currentMax={myRange[1]}
  onChange={setMyRange}
/>
```

File: packages/es-components/src/components/controls/range-selector/RangeSelector.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+  const [minPercentage, setMinPercentage] = useState(initStateMin);
+  const [maxPercentage, setMaxPercentage] = useState(initStateMax);
```
Comment (stevematney) [inline packages/es-components/src/components/controls/range-selector/RangeSelector.js:80]: Since we're not doing anything other than rendering them out, these percentages don't need to be in state. They can just be derived inline from `min` and `max`.

File: packages/es-components/src/components/controls/range-selector/RangeSelector.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+};
+
+RangeSelector.defaultProps = {
+  currentMinValue: 2500,
+  currentMaxValue: 7500,
+  minValue: 0,
+  maxValue: 10000,
+  progressColor: '#17A2B8',
+  sliderColor: '#ddd',
+  thumbColor: '#17A2B8',
```
Comment (stevematney) [inline packages/es-components/src/components/controls/range-selector/RangeSelector.js:176]: All these color defaults should be derived from theme values, rather than hard-coded.

**Review by stevematney — CHANGES_REQUESTED** [review]
This all looks good! Thanks for submitting it!
Overall, I like that the code is pretty simple. I just have a few minor changes to hopefully make it even simpler.

Also, would you mind including a screenshot of what this looks like in your PR?

File: packages/es-components/src/components/controls/range-selector/RangeSelector.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        <RangeSelectorProgress
+          minPercentage={minPercentage}
+          maxPercentage={maxPercentage}
+          progressColor={progressColor}
+        ></RangeSelectorProgress>
+      </RangeSelectorSlider>
+      <RangeSelectorInputContainer>
+        <RangeSelectorInput
+          type="range"
+          aria-label="min-input"
```
Comment (stevematney) [inline packages/es-components/src/components/controls/range-selector/RangeSelector.js:125]: `aria-label` values should be in plain English, as they'll often be read aloud by Assistive Technology.
```suggestion
          aria-label="Minimum Value Input"
```

File: packages/es-components/src/components/controls/range-selector/RangeSelector.js [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+          max={maxValue}
+          value={min}
+          onChange={changeMin}
+          onMouseUp={changeMinEnd}
+          progressColor={progressColor}
+          thumbColor={thumbColor}
+        ></RangeSelectorInput>
+        <RangeSelectorInput
+          type="range"
+          aria-label="max-input"
```
Comment (stevematney) [inline packages/es-components/src/components/controls/range-selector/RangeSelector.js:136]:
```suggestion
          aria-label="Maximum Value Input"
```

File: packages/es-components/src/components/controls/range-selector/RangeSelector.js (hunk 1 above)
Comment (DoChunyu) [reply packages/es-components/src/components/controls/range-selector/RangeSelector.js:78]: the  parent will give min, max and set current min and current max.   so it is not internal controlled, 
image

File: packages/es-components/src/components/controls/range-selector/RangeSelector.js (hunk 2 above)
Comment (DoChunyu) [reply packages/es-components/src/components/controls/range-selector/RangeSelector.js]: removed minPercentage  maxPercentage state, make it derived.

File: packages/es-components/src/components/controls/range-selector/RangeSelector.js (hunk 5 above)
Comment (DoChunyu) [reply packages/es-components/src/components/controls/range-selector/RangeSelector.js]: change label to aria-label="Maximum Value Input"

File: packages/es-components/src/components/controls/range-selector/RangeSelector.js (hunk 4 above)
Comment (DoChunyu) [reply packages/es-components/src/components/controls/range-selector/RangeSelector.js]: change label to  Minimum Value Input

**Comment by DoChunyu** [comment]
> 
Here is screenshot: 
image

File: packages/es-components/src/components/controls/range-selector/RangeSelector.js (hunk 3 above)
Comment (DoChunyu) [reply packages/es-components/src/components/controls/range-selector/RangeSelector.js]: changed default color to theme color:
progressColor: props => props.theme.colors.success,
  sliderColor: props => props.theme.colors.gray2 ,
  thumbColor: props => props.theme.colors.primary,

File: packages/es-components/src/components/controls/range-selector/RangeSelector.js (hunk 1 above)
Comment (DoChunyu) [reply packages/es-components/src/components/controls/range-selector/RangeSelector.js]: changed from duo range selector to single range selector, removed state in the component.

**Comment by DoChunyu** [comment]
image

**Comment by DoChunyu** [comment]
at this moment, nobody use the slider, customization team think they may build slider  later after they finalized UX design.
